src/components/withDarkMode.js

```diff
@@ -9,8 +9,16 @@ const withDarkMode = (ComponentToWrap) => (props) => {
   }, []);
   const isDarkDefault = false;
   const checkLocalStorage = isLocalStorageAvailable();
-  const darkModeOptions = checkLocalStorage ? {} : { storageProvider: false };
-  const darkMode = useDarkMode(isDarkDefault, darkModeOptions);
+  const darkMode = () => {
+    if (checkLocalStorage) {
+      const localStorageTheme = localStorage.getItem('darkMode');
+      if (localStorageTheme === 'true' || localStorageTheme === 'false') {
+        return JSON.parse(localStorageTheme);
+      }
+    } else {
+      return useDarkMode(isDarkDefault, { storageProvider: false });
```

Comment:
this breaks a rule of hooks, [hooks shouldn't be called conditionally](https://legacy.reactjs.org/docs/hooks-rules.html). you could try calling this hook above the conditional, store the result in variable, and if we hit this branch just return that result. not sure how that would interact with the value in localStorage though